lib/issuer.js

```diff
@@ -245,6 +240,59 @@ api.claim = function(actor, id, callback) {
   });
 };
 
+function getCredentialSigningKey(credential, callback) {
+  async.auto({
+    getIssuer: function(callback) {
+      var issuer = credential.issuer;
+      brIdentity.get(null, issuer, function(err, results) {
+        callback(err, results);
+      });
+    },
+    getPreferredSigningKey: ['getIssuer', function(callback, results) {
+      var identity = results.getIssuer;
+      var keyId = null;
+
+      if(identity.sysPreferences && identity.sysPreferences.credentialSigningKey) {
+        keyId = identity.sysPreferences.credentialSigningKey;
+      }
+      if(!keyId) {
+        return callback();
+      }
+      brKey.getPublicKey({id: keyId}, identity,
+        function(err, publicKey, meta, privateKey) {
+        publicKey.privateKey = privateKey;
+        callback(err, publicKey);
+      });
+    }],
+    getSigningKey: ['getPreferredSigningKey', function(callback, results) {
+      if(results.getPreferredSigningKey) {
+        // Use the preferred signing key
+        return callback(null, results.getPreferredSigningKey);
+      }
+      var identity = results.getIssuer;
+      // If no preferred signing key, get their first non-revoked public key
+      brKey.getPublicKeys(identity.id, identity, function(err, results) {
```

Comment:
Need to add an "action" for setting the preferred key to the keys view for key rows.